pkg/render/fluentd.go

```diff
@@ -466,21 +473,40 @@ func (c *fluentdComponent) daemonset() *appsv1.DaemonSet {
 		initContainers = append(initContainers, c.cfg.FluentdKeyPair.InitContainer(LogCollectorNamespace))
 	}
 
-	podTemplate := relasticsearch.DecorateAnnotations(&corev1.PodTemplateSpec{
-		ObjectMeta: metav1.ObjectMeta{
-			Annotations: annots,
-		},
-		Spec: corev1.PodSpec{
-			NodeSelector:                  map[string]string{},
-			Tolerations:                   c.tolerations(),
-			ImagePullSecrets:              secret.GetReferenceList(c.cfg.PullSecrets),
-			TerminationGracePeriodSeconds: &terminationGracePeriod,
-			InitContainers:                initContainers,
-			Containers:                    []corev1.Container{c.container()},
-			Volumes:                       c.volumes(),
-			ServiceAccountName:            c.fluentdNodeName(),
-		},
-	}, c.cfg.ESClusterConfig, c.cfg.ESSecrets).(*corev1.PodTemplateSpec)
+	var podTemplate *corev1.PodTemplateSpec
+	if c.cfg.EnterpriseVariant {
```

Comment:
We don't need this duplication. The pod templates are the same, the only difference is the decorator.

let's do this instead:

```go
podTemplate = &corev1.PodTemplateSpec{ ... }

if enterprise {
  podTempalte = relasticsearch.DecorateAnnotations(podTemplate)
}